app/models/apple/podcast_container.rb

```diff
@@ -29,76 +27,103 @@ def self.update_podcast_container_file_metadata(api, episodes_to_sync)
+      episodes_to_create = episodes.reject { |ep| ep.podcast_container.present? }
 
       new_containers_response =
         api.bridge_remote_and_retry!("createPodcastContainers",
                                      create_podcast_containers_bridge_params(api, episodes_to_create))
 
-      show.reload
-
-      # Make sure we have local copies of the remote metadata At this point and
-      # errors should be resolved and we should have then intended set of
-      # resources created.
-
-      episodes_by_id = episodes_to_create.map { |ep| [ep.id, ep] }.to_h
-
-      new_containers_response.map do |row|
-        ep = episodes_by_id.fetch(row["request_metadata"]["apple_episode_id"])
-
-        create_podcast_container(ep, row)
+      join_on_apple_episode_id(episodes_to_create, new_containers_response) do |ep, row|
+        upsert_podcast_container(ep, row)
       end
     end
 
-    def self.create_podcast_container(ep, row)
+    def self.upsert_podcast_container(episode, row)
       external_id = row.dig("api_response", "val", "data", "id")
 
       pc =
-        if pc = where(episode_id: ep.feeder_id,
-                      apple_episode_id: ep.apple_id,
-                      vendor_id: ep.audio_asset_vendor_id,
+        if pc = where(episode_id: episode.feeder_id,
+                      apple_episode_id: episode.apple_id,
+                      vendor_id: episode.audio_asset_vendor_id,
                       external_id: external_id).first
-          pd.update!(api_response: row)
+          pc.update(api_response: row, updated_at: Time.now.utc)
+          Rails.logger.info("Updating local podcast container w/ Apple id #{external_id} for episode #{episode.feeder_id}")
           pc
         else
-          create!(episode_id: ep.feeder_id,
-                  external_id: external_id,
-                  vendor_id: ep.audio_asset_vendor_id,
-                  apple_episode_id: ep.apple_id,
-                  source_url: ep.enclosure_url,
-                  source_filename: ep.enclosure_filename,
-                  api_response: row)
+          pc = create!(episode_id: episode.feeder_id,
+                       external_id: external_id,
+                       vendor_id: episode.audio_asset_vendor_id,
+                       apple_episode_id: episode.apple_id,
+                       source_url: episode.enclosure_url,
+                       source_filename: episode.enclosure_filename,
+                       api_response: row)
+          Rails.logger.info("Creating local podcast container w/ Apple id #{external_id} for episode #{episode.feeder_id}")
+          pc
         end
 
       SyncLog.create!(feeder_id: pc.id, feeder_type: :podcast_containers, external_id: external_id)
 
       pc
     end
 
-    def self.get_podcast_containers(api, episodes_to_sync)
-      api.bridge_remote_and_retry!("getPodcastContainers", get_podcast_containers_bridge_params(api, episodes_to_sync))
+    def self.get_podcast_containers_via_episodes(api, episodes)
```

Comment:
Following the hypermedia links here (and in the case of delivery / delivery_files).

Get the remote container resource _nested under the episode_. Flatten and follow those links to the canonical resource, the base podcast container resource in this case. Then update the local state.

This adopts the pattern, that if we're storing remote state locally, always store the non-enumerated "base" resource instead of the nested resource.  e.g. store `podcastContainers/1234` instead of `podcastContainers?filter[vendorId]=<some episode id>`